rcl/fibonacci_action_client/main.c

```diff
@@ -0,0 +1,165 @@
+#include <rcl/rcl.h>
+#include <rcl_action/rcl_action.h>
+#include <rcl/error_handling.h>
+#include <example_interfaces/action/fibonacci.h>
+
+#include <stdio.h>
+#include <unistd.h>
+#include <pthread.h>
+
+int main(int argc, const char * const * argv)
+{
+  rcl_ret_t rv;
+
+  rcl_init_options_t options = rcl_get_zero_initialized_init_options();
+  rv = rcl_init_options_init(&options, rcl_get_default_allocator());
+  if (RCL_RET_OK != rv) {
+    printf("rcl init options error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  rcl_context_t context = rcl_get_zero_initialized_context();
+  rv = rcl_init(argc, argv, &options, &context);
+  if (RCL_RET_OK != rv) {
+    printf("rcl initialization error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  rcl_node_options_t node_ops = rcl_node_get_default_options();
+  rcl_node_t node = rcl_get_zero_initialized_node();
+  rv = rcl_node_init(&node, "fibonacci_action_client_rcl", "", &context, &node_ops);
+  if (RCL_RET_OK != rv) {
+    printf("Node initialization error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  const char * action_name = "fibonacci";
+  const rosidl_action_type_support_t * action_type_support = ROSIDL_GET_ACTION_TYPE_SUPPORT(example_interfaces, Fibonacci);
+  rcl_action_client_t action_client = rcl_action_get_zero_initialized_client();
+  rcl_action_client_options_t action_client_ops = rcl_action_client_get_default_options();
+
+  rv = rcl_action_client_init(
+      &action_client,
+      &node,
+      action_type_support,
+      action_name,
+      &action_client_ops);
+
+  if (RCL_RET_OK != rv) {
+    printf("Action server initialization error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  size_t num_subscriptions, num_guard_conditions, num_timers, num_clients, num_services;
+
+  rcl_action_client_wait_set_get_num_entities(&action_client, &num_subscriptions, &num_guard_conditions, &num_timers, &num_clients, &num_services);
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  rv = rcl_wait_set_init(&wait_set, num_subscriptions, num_guard_conditions, num_timers, num_clients, num_services, 0, &context, rcl_get_default_allocator());
+  if (RCL_RET_OK != rv) {
+    printf("Wait set initialization error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  int64_t goal_sequence_number;
+  int64_t result_sequence_number;
+
+  bool done = false;
+  bool goal_accepted = false;
+  int order = 10;
+
+  sleep(2);
```

Comment:
Why do you sleep here?

Reply:
Because it does not seems to work if I send the request inmmediatly after initialization stage.

Reply:
Weird, please add a comment clarifying it.

Reply:
Done